pkg/proxy/endpointslicecache.go

```diff
@@ -66,26 +65,10 @@ type endpointSliceTracker struct {
 // corresponding EndpointSlices.
 type endpointSliceDataByName map[string]*endpointSliceData
 
-// endpointSliceData contains just the attributes kube-proxy cares about.
-// Used for caching. Intentionally small to limit memory util.
+// endpointSliceData contains information about a single EndpointSlice update or removal.
 type endpointSliceData struct {
-	Ports     []discovery.EndpointPort
-	Endpoints []*endpointData
-	Remove    bool
-}
-
-// endpointData contains just the attributes kube-proxy cares about.
-// Used for caching. Intentionally small to limit memory util.
-// Addresses, NodeName, and Zone are copied from EndpointSlice Endpoints.
-type endpointData struct {
-	Addresses []string
-	NodeName  *string
-	Zone      *string
-	ZoneHints sets.Set[string]
-
-	Ready       bool
-	Serving     bool
-	Terminating bool
+	endpointSlice *discovery.EndpointSlice
```

Comment:
It's fine. Objects in the informer cache are const; when it gets an update it will create a new object.

Reply:
hmm, the informer cache Store has Update operations, AFAIK objects can be updated

Reply:
You can update the _cache_, but that means replacing objects within it, not modifying them.

Maybe this isn't really documented? But note that our e2e tests run with `KUBE_CACHE_MUTATION_DETECTOR` set so that they will panic if somebody tries to modify an object owned by an informer.

Reply:
> Objects in the informer cache are const

They must be treated that way, but it's not enforced (other than in a best-effort test way via `KUBE_CACHE_MUTATION_DETECTOR`). The informer / client-go machinery does not modify objects inside cache.Store

> when it gets an update it will create a new object

that's correct, Store#Update replaces a given instance of an object in the cache with a new instance of the same object observed via an update watch event

> Maybe this isn't really documented?

It's sort of documented for the handler functions, but not super visibly:

```
// ResourceEventHandler can handle notifications for events that
// happen to a resource. The events are informational only, so you
// can't return an error.  The handlers MUST NOT modify the objects
// received; this concerns not only the top level of structure but all
// the data structures reachable from it.
```